sample-app/src/components/InputDropdown.tsx

```diff
@@ -85,37 +85,67 @@ export default function InputDropdown({
   const inputRef = useRef<HTMLInputElement>(null);
   const dropdownRef = useRef<HTMLDivElement>(null);
   const inputDropdownRef = useRef<HTMLDivElement>(null);
-
   if (!shouldDisplayDropdown && screenReaderKey) {
     setScreenReaderKey(0);
   }
 
+  const isMounted = useRef<boolean>(false);
+  useEffect(() => {
```

Comment:
Was this needed by something added by this PR or was always an issue with InputDropdown? Also do you know in what kinds of situations we get state updates on unmounted components?

Reply:
I believe a component unmount when the parent component is no longer rendered or the parent component performs an update that does not render this instance (or calling ReactDOM.unmountComponentAtNode). 
This would be a potential issue if a user's custom component that uses InputDropdown does some custom render logic or include some special logic in `option.onClick?.();` the would trigger unmount before calling the following state update functions. 
In the case of this PR, `option.onClick?.();` happens to perform a route navigation necessary for a vertical link option, so the following state update functions would be attempting to update state for an unmounted component.

Reply:
would moving the onClick after setLatestUserInput and dispatch help?

Reply:
I think it would help? The error appears sporadically, I tested it a couple times with the order swapped and i'm not seeing the error at the moment so it seems fine. However, if I remember correctly, the setState/dispatch stuff does not guarantee unsubscriptions until next dispatch according to some caveat in redux source code so I think there's a chance it could still cause an error? which is why I kept it in this format and a bit hesitant in just swapping the order.

Reply:
was the stack trace you were getting pointing to anything in redux? I thought that the dispatch and setLatestUserInput here would have nothing (directly) to do with our redux store

Reply:
the stack trace doesn't have point related to it. It was similar to the one we encountered before:
![Screen Shot 2021-12-09 at 3 14 28 PM](https://user-images.githubusercontent.com/36055303/145469102-09115c30-8383-44fa-90cd-b0772b735f3f.png)

Reply:
update: I swapped the order since I cannot replicate the error anymore and assuming that react state update subscription doesn't have the same caveat as redux.